From 2ce1aed6d8980ae341cd3499d8efc36ce9b2ac5d Mon Sep 17 00:00:00 2001
From: Reini Urban <rurban@cpanel.net>
Date: Fri, 4 Apr 2014 12:59:31 -0500
Subject: [PATCH 5/5] C: fix make -jn test, parallel tests (Ticket #85336)

strip make jobserver-fds from MAKEFLAGS
print MAKE* ENV values with failed make command
---
 C/C.pm    | 10 +++++++++-
 C/Changes |  6 ++++++
 Changes   |  3 +--
 3 files changed, 16 insertions(+), 3 deletions(-)

diff --git C/C.pm C/C.pm
index 21f7dfe..0dfa73b 100644
--- C/C.pm
+++ C/C.pm
@@ -812,12 +812,14 @@ sub make {
     my ($o) = @_;
     my $make = $o->{ILSM}{MAKE} || $Config::Config{make}
       or croak "Can't locate your make binary";
+    local $ENV{MAKEFLAGS} = $ENV{MAKEFLAGS} =~ s/(--jobserver-fds=[\d,]+)//;
     $o->system_call("$make", 'out.make');
 }
 sub make_install {
     my ($o) = @_;
     my $make = $o->{ILSM}{MAKE} || $Config::Config{make}
       or croak "Can't locate your make binary";
+    local $ENV{MAKEFLAGS} = $ENV{MAKEFLAGS} =~ s/(--jobserver-fds=[\d,]+)//;
     $o->system_call("$make pure_install", 'out.make_install');
 }
 sub cleanup {
@@ -864,7 +866,7 @@ sub build_error_message {
     }
 
     my $errcode = $? >> 8;
-    return $output . <<END;
+    $output .= <<END;
 
 A problem was encountered while attempting to compile and install your Inline
 $o->{API}{language} code. The command that failed was:
@@ -876,6 +878,12 @@ $build_dir
 To debug the problem, cd to the build directory, and inspect the output files.
 
 END
+    if ($cmd =~ /^make >/) {
+      for (sort keys %ENV) {
+        $output .= "$_ = $ENV{$_}\n" if /^MAKE/;
+      }
+    }
+    return $output;
 }
 
 #==============================================================================
diff --git C/Changes C/Changes
index f3406ac..0019ec6 100644
--- C/Changes
+++ C/Changes
@@ -1,5 +1,11 @@
 Revision history for Perl extension Inline::C.
 
+--- version 0.54_01
+date 2014-04-04
+     fix make -jn test, parallel tests (Ticket #85336) (rurban)
+     disable BUILD_NOISY redirects on MSWin32 with cmd.exe (rurban)
+     print exitcode with failed commands and MAKE* ENV values (rurban)
+
 0.54 Sat 29 mar 2014
      Update version number to 0.54
      Release version 0.54 to CPAN
diff --git Changes Changes
index 641e8ba..c128032 100644
--- Changes
+++ Changes
@@ -1,11 +1,10 @@
 # Revision history for Perl extension Inline.
 
 --- version 0.54_01
-date 2014-04-04 12:22:37
+date 2014-04-04
      import Fcntl constants for flock (rurban)
      flock only on supported platforms, would die on VMS,riscos,VOS (rurban)
      print exitcode with failed commands
-     disable BUILD_NOISY redirects on MSWin32 with cmd.exe
 
 --- version 0.54
 date Sat 29 Mar 2014
-- 
1.9.1

